### ISSUE TITLE
Not working on macos M1 chip.

### ISSUE THREAD

**Issue by Aganthor** [verpeteren/rust-simd-noise#52]
Hello!

This crate can't compile on a MacBook air with the M1 CPU.

Example of some of the error message:
`error[E0425]: cannot find function, tuple struct or tuple variant `F32x4` in this scope
   --> /Users/luc/.cargo/registry/src/github.com-1ecc6299db9ec823/simdnoise-3.1.6/src/sse41.rs:365:9
    |
365 |         F32x4(gain),
    |         ^^^^^ not found in this scope`

I've read that the M1 chip supports SIMD, but only the Neon instruction sets. Is this why it's not working?

Thanks in advance.

**Comment by jackmott** [comment]
Correct, currently only intel/amd SIMD variants are supported. It is possible to add support for NEON, just a lot of grunt work.

**Comment by jackmott** [comment]
And its the Simdeez library that needs to add support, then it would more or less "just work" in this noise library: https://github.com/arduano/simdeez

**Comment by valadaptive** [comment]
Even if platform-specific SIMD implementations are absent, shouldn't the code fall back to the scalar implementation rather than failing to compile?

**Comment by verpeteren** [comment]
@valadaptive: It is working indeed (on `m2`, falling back to scalar.
@thomasza92: Indeed, `simdeez` now also has the `neon` instructions ready.
I made a [start](https://github.com/verpeteren/rust-simd-noise/pull/58) to prepare simdnoise for neon, stay tuned.
